pytorch_lightning/strategies/tpu_spawn.py

```diff
@@ -267,8 +247,7 @@ def _wrapped_function(
 
     def _worker_setup(self, process_idx: int):
         reset_seed()
-        self.tpu_local_core_rank = xm.get_local_ordinal()
-        self.tpu_global_core_rank = xm.get_ordinal()
+        self.set_world_ranks(process_idx)
```

Comment:
Should we introduce a TPUClusterEnv ?

Reply:
If there is a different way of launching tpu jobs than currently, then probably yes. If not, I don't see what it would do differently than the LightningEnvironment.

Reply:
The ENV variable names used by TPU are different, so I would expect the code to be cleaner if fully encapsulated under a TPUClusterEnvironement.